### PR TITLE
Added context support for storage methods

### PR DESCRIPTION
**Pull request by Drynok**

### CODE REVIEW

**Comment by protsack-stephan** [comment]
We need to create interfaces for those new methods, also extend `Storage` interface and add unit tests for this.

File: lib/fs/storage.go [hunk 1]
```diff
@@ -51,6 +52,11 @@ func (s Storage) List(path string, options ...map[string]interface{}) ([]string,
 	return d.Readdirnames(-1)
 }
 
+// ListWithContext reads the path content
+func (s Storage) ListWithContext(ctx context.Context, path string, options ...map[string]interface{}) ([]string, error) {
```
Comment (protsack-stephan) [inline lib/fs/storage.go]: If the `ctx` param is not used let's replace it with `_`.

File: lib/fs/storage.go [hunk 2]
```diff
@@ -80,6 +86,11 @@ func (s Storage) Walk(path string, callback func(path string)) error {
 	})
 }
 
+// WalkWithContext recursively look for files in directory
+func (s Storage) WalkWithContext(ctx context.Context, path string, callback func(path string)) error {
```
Comment (protsack-stephan) [inline lib/fs/storage.go]: If the `ctx` param is not used let's replace it with `_`.

File: lib/fs/storage.go [hunk 3]
```diff
@@ -113,6 +124,11 @@ func (s Storage) Get(path string) (io.ReadCloser, error) {
 	return os.Open(loc)
 }
 
+// GetWithContext get object from storage
+func (s Storage) GetWithContext(ctx context.Context, path string) (io.ReadCloser, error) {
```
Comment (protsack-stephan) [inline lib/fs/storage.go]: If the `ctx` param is not used let's replace it with `_`.

File: lib/fs/storage.go [hunk 4]
```diff
@@ -141,6 +157,11 @@ func (s Storage) Put(path string, body io.Reader) error {
 	return ioutil.WriteFile(loc, buff, 0766)
 }
 
+// PutWithContext object into storage
+func (s Storage) PutWithContext(ctx context.Context, path string, body io.Reader) error {
```
Comment (protsack-stephan) [inline lib/fs/storage.go]: If the `ctx` param is not used let's replace it with `_`.

File: lib/fs/storage.go [hunk 5]
```diff
@@ -157,6 +178,11 @@ func (s *Storage) Delete(path string) error {
 	return os.Remove(loc)
 }
 
+// DeleteWithContext remove object from storage
+func (s *Storage) DeleteWithContext(ctx context.Context, path string) error {
```
Comment (protsack-stephan) [inline lib/fs/storage.go]: If the `ctx` param is not used let's replace it with `_`.

File: pkg/storage/storage.go [hunk 6]
```diff
@@ -22,11 +28,20 @@ type Lister interface {
 	List(path string, options ...map[string]interface{}) ([]string, error)
 }
 
+type ListerWithContext interface {
```
Comment (protsack-stephan) [inline pkg/storage/storage.go:32]: Please add comment explaining what this interface for.

File: pkg/storage/storage.go [hunk 7]
```diff
@@ -37,11 +52,21 @@ type Getter interface {
 	Get(path string) (io.ReadCloser, error)
 }
 
+// Getter get object from storage
```
Comment (protsack-stephan) [inline pkg/storage/storage.go]: This should be `GetterWithContext` instead of `Getter`.

File: pkg/storage/storage.go [hunk 8]
```diff
@@ -52,6 +77,11 @@ type Deleter interface {
 	Delete(path string) error
 }
 
+// Deleter delete object from storage
```
Comment (protsack-stephan) [inline pkg/storage/storage.go]: This should be `DeleterWithContext` instead of `Deleter`.

**Review by protsack-stephan — COMMENTED** [review]
LGTM! Couple of minor comments.

File: lib/fs/storage.go (hunk 1 above)
Comment (Drynok) [reply lib/fs/storage.go]: fixed

File: lib/fs/storage.go (hunk 2 above)
Comment (Drynok) [reply lib/fs/storage.go]: fixed

File: pkg/storage/storage.go (hunk 8 above)
Comment (Drynok) [reply pkg/storage/storage.go]: fixed

File: lib/fs/storage.go (hunk 5 above)
Comment (Drynok) [reply lib/fs/storage.go]: fixed

**Pull request merged**